src/plugins/template/src/async_infer_request.cpp

```diff
@@ -51,3 +54,10 @@ ov::template_plugin::AsyncInferRequest::~AsyncInferRequest() {
     ov::IAsyncInferRequest::stop_and_wait();
 }
 // ! [async_infer_request:dtor]
+
+// ! [async_infer_request:cancel]
+void ov::template_plugin::AsyncInferRequest::cancel() {
+    ov::IAsyncInferRequest::cancel();
+    m_cancel_callback();
```

Comment:
1. why do we need to store callback instead of simple `request->terminate();` ?
2. why do we need to call base class?

Reply:
1. Because it is simpler than hold one more pointer to template infer request
2. Because the base method change the status of infer request which is used in some common functions